### PR TITLE
Use HTML parser to scan the file type as it's shown to user

### PR DESCRIPTION
**Pull request by babenek**
## Description

- Add HTML files scan in text representation and tables research. Line numbers may be wrong.

## How has this been tested?

- [x] UnitTests
- [x] place a password in table with named header "password" - the password should be found

### CODE REVIEW

File: credsweeper/common/constants.py [hunk 1]
```diff
@@ -110,4 +110,4 @@ class DiffRowType(Enum):
 AVAILABLE_ENCODINGS = (DEFAULT_ENCODING, "utf16", "latin_1")
 
 # to limit memory usage in case of recursive scan
-RECURSIVE_SCAN_LIMITATION = 1 << 30
+RECURSIVE_SCAN_LIMITATION = 1 << 32
```
Comment (babenek) [inline credsweeper/common/constants.py:113]:
```suggestion
RECURSIVE_SCAN_LIMITATION = 1 << 30
```
rollback

Comment (babenek) [inline credsweeper/common/constants.py:113]:
```suggestion
RECURSIVE_SCAN_LIMITATION = 1 << 30
```
rollback

**Review by csh519 — COMMENTED** [review]
Thank you for implment html parser feature.
And I've checked it and working good.

BTW I think we need some refactoring for `DeepContentProvider`.
It's too huge now and `app.py` has a lot of if-else chain 😢
Let's think about refactoring next time.

LGTM 👍

**Review by csh519 — APPROVED** [review]

**Review by iuriimet — APPROVED** [review]

**Comment by babenek** [comment]
> Thank you for implment html parser feature. And I've checked it and working good.
> 
> BTW I think we need some refactoring for `DeepContentProvider`. It's too huge now and `app.py` has a lot of if-else chain cry Let's think about refactoring next time.
> 
> LGTM +1

https://github.com/Samsung/CredSweeper/pull/270 - some refactoring was made.

**Pull request merged**